### ISSUE TITLE
Contributing to forge: Forked project crashes with java.util.NoSuchElementException and java.lang.reflect.InvocationTargetException

### ISSUE THREAD

**Issue by thedead2** [MinecraftForge/MinecraftForge#9719]
**Minecraft Version:** 1.20.1
**Forge Version:**  Idk, it’s the current 1.20.x branch
**IDE:** IntelliJ IDEA 2023.2.2 (Community Edition)
**JDK/ Java Version:** jbr-17 (17.0.8)

### **Description of the issue:**

I forked the 1.20.x branch of the MinecraftForge repository to create a pull request for #9717. I followed the guides provided [here](https://docs.minecraftforge.net/en/latest/forgedev/) and [here](https://github.com/MinecraftForge/MinecraftForge/wiki/If-you-want-to-contribute-to-Forge). After cloning and importing the project into IntelliJ, I run the setup and genIntelliJRuns tasks as described. They both completed successfully. To test whether the setup of the project was successful I tried to run the clean client, fmlonly client and forge client run configurations created by the genIntelliJRuns task. Clean client and fmlonly client worked as intended but when I tried to run the forge client configuration the game crashes  directly after start up of the mod discovering, with `java.util.NoSuchElementException` at `net.minecraftforge.fml.earlydisplay.DisplayWindow.updateModuleReads(DisplayWindow.java:584)`. 
It seems like the module "forge" isn’t present. I also tried running the forge server run configuration but it crashes with `java.lang.reflect.InvocationTargetException` at `net.minecraftforge.fml.loading.targets.CommonLaunchHandler.runTarget(CommonLaunchHandler.java:111)`.
As they were both exceptions in thread main they aren’t included inside the debug logs and there are no crash reports. I will still provide the logs but for the exceptions themself I can only supply screenshots.

Clearly I must have done something wrong when setting the project up but I can't seem to figure out what it was. I have checked inside the project structure of IntelliJ if for some reason the IDE didn't recognised a module, but it seems all perfectly fine. From the exceptions I would say that I’m missing mandatory modules such as „forge“ and „minecraft“, which definitely shouldn’t be the case.
I tried forking, cloning and setting up the project three times now, always with the same exceptions on runtime.
Hope someone can help me. I couldn’t find any similar cases in the forums or issues, therefor I’m  creating this issue.

**Steps to Reproduce:**
 1. For me: Start "forge client" or "forge server" run configuration.

**Logs:**
https://gist.github.com/thedead2/50cd2aafa74c7bbb3f8228794083d1a4 https://gist.github.com/thedead2/23828fcb8b031876e6e0208e855a57b0

**Screenshots:**

<img width="1320" alt="Screenshot 2023-09-21 at 20 02 03" src="https://github.com/MinecraftForge/MinecraftForge/assets/62204843/8a8cbc8f-71fa-4cf0-8217-d4e58c098d6f">

<img width="1320" alt="Screenshot 2023-09-21 at 20 00 43" src="https://github.com/MinecraftForge/MinecraftForge/assets/62204843/66ed5f6b-1835-42f0-ac6d-09950baceb6f">

**Comment by LexManos** [comment]
Interesting I cant give any advice for intellij because I dont use it. But as it's a crash during the early loading screen you can try disabling that by setting the provider to "none" in fml.toml

Also be sure to do a clean and regen your run configs in your ide if you make any local commits or anything because the version numbers and paths are really sensitive

**Comment by thedead2** [comment]
Hi, thanks for the reply. Setting the provider to "none" only shifted the exception to `java.lang.reflect.InvocationTargetException` at `net.minecraftforge.fml.loading.targets.CommonLaunchHandler.runTarget(CommonLaunchHandler.java:111)`.

But I tried setting up the project for a forth time. This time it worked. I didn't clone and import the project directly to IntelliJ, instead I cloned the repository via the GitHub Desktop Client and afterwards opened the project with IntelliJ. After running the setup and genIntelliJRuns tasks the forge client configuration started without any problems. Seems to me like IntelliJ did something weird when importing the project from git?!

So just in case someone has the same issue: Don't use IntelliJ to clone and import the project!
